frame/plasm-lockdrop/src/lib.rs

```diff
@@ -0,0 +1,662 @@
+        /// List of lockdrop authority id's.
+        Keys get(fn keys): Vec<T::AuthorityId>;
+        /// Token claim requests.
+        Claims get(fn claims): linked_map hasher(blake2_256)
+                               ClaimId => Claim;
+        /// Lockdrop alpha parameter.
+        Alpha get(fn alpha) config(): u128;
+        /// Lockdrop dollar rate parameter.
+        DollarRate get(fn dollar_rate) config(): T::DollarRate;
+        /// Lockdrop dollar rate median filter table.
+        DollarRateF get(fn dollar_rate_f): linked_map hasher(blake2_256)
+                                           T::AccountId => (T::Moment, T::DollarRate);
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        /// Initializing events
+        fn deposit_event() = default;
+
+        /// Clean the state on initialisation of a block
+        fn on_initialize(_now: T::BlockNumber) {
+            // At the beginning of each block execution, system triggers all
+            // `on_initialize` functions, which allows us to set up some temporary state or - like
+            // in this case - clean up other states
+            <Requests>::kill();
+        }
+
+        /// Request authorities to check locking transaction. 
+        #[weight = SimpleDispatchInfo::FixedNormal(1_000_000)]
+        fn request(
+            origin,
+            params: Lockdrop,
+        ) {
+            let _ = ensure_signed(origin)?;
+            let claim_id = BlakeTwo256::hash_of(&params);
+            ensure!(!<Claims>::get(claim_id).complete, "claim should not be already paid"); 
+
+            if !<Claims>::exists(claim_id) {
+                let amount = match params {
+                    Lockdrop::Bitcoin { value, duration, .. } => {
+                        // Average block duration in BTC is 10 min = 600 sec
+                        let duration_sec = duration * 600;
+                        // Cast bitcoin value to PLM order:
+                        // satoshi = BTC * 10^9;
+                        // PLM unit = PLM * 10^15;
+                        // (it also helps to make evaluations more precise)
+                        let value_btc = (value as u128) * 1_000_000;
+                        Self::issue_amount(value_btc, duration_sec.into())
+                    }
+                };
+                let claim = Claim { params, amount, .. Default::default() };
+                <Claims>::insert(claim_id, claim);
+            }
+
+            <Requests>::mutate(|requests| requests.push(claim_id));
+            Self::deposit_event(RawEvent::ClaimRequest(claim_id));
+        }
+
+        /// Claim tokens according to lockdrop procedure.
+        #[weight = SimpleDispatchInfo::FixedNormal(1_000_000)]
+        fn claim(
+            origin,
+            claim_id: ClaimId,
+        ) {
+            let _ = ensure_signed(origin)?;
+            ensure!(T::Time::now() < T::LockdropEnd::get(), "lockdrop should be finished");
```

Comment:
If we can not claim after Lockdrop, that is very severe. Because they may be going to rash demand.
I think. `request` method should have this condition.

Reply:
Sure, this wrong, we can claim at any time because `total_issue` isn't more required. Fixed.